ParserHelpers.h

```diff
@@ -14,6 +15,12 @@ void ignoreItem(const std::string& unused, std::istream& theStream);
 void ignoreObject(const std::string& unused, std::istream& theStream);
 void ignoreString(const std::string& unused, std::istream& theStream);
 
+template <typename T>
+// function template only enabled for integer types
```

Comment:
Maybe also add comments demonstrating the use:

```
char stringToInteger<char>(const std::string& str, bool skipPartialMatchWarning);
unsigned char stringToInteger<unsigned char>(const std::string& str, bool skipPartialMatchWarning);
short stringToInteger<short>(const std::string& str, bool skipPartialMatchWarning);
unsigned short stringToInteger<unsigned short>(const std::string& str, bool skipPartialMatchWarning);
int stringToInteger<int>(const std::string& str, bool skipPartialMatchWarning);
unsigned int stringToInteger<unsigned int>(const std::string& str, bool skipPartialMatchWarning);
long stringToInteger<long>(const std::string& str, bool skipPartialMatchWarning);
unsigned long stringToInteger<unsigned long>(const std::string& str, bool skipPartialMatchWarning);
long long stringToInteger<long long>(const std::string& str, bool skipPartialMatchWarning);
unsigned long long stringToInteger<unsigned long long>(const std::string& str, bool skipPartialMatchWarning);
```

Reply:
Also added an example with unsigned int
```
* converts string to integer types, e.g.: unsigned int value = stringToInteger<unsigned int>("420");
* works for:
* signed char stringToInteger<signed char>(const std::string& str, bool skipPartialMatchWarning);
* unsigned char stringToInteger<unsigned char>(const std::string& str, bool skipPartialMatchWarning);
* short stringToInteger<short>(const std::string& str, bool skipPartialMatchWarning);
* unsigned short stringToInteger<unsigned short>(const std::string& str, bool skipPartialMatchWarning);
* int stringToInteger<int>(const std::string& str, bool skipPartialMatchWarning);
* unsigned int stringToInteger<unsigned int>(const std::string& str, bool skipPartialMatchWarning);
* long stringToInteger<long>(const std::string& str, bool skipPartialMatchWarning);
* unsigned long stringToInteger<unsigned long>(const std::string& str, bool skipPartialMatchWarning);
* long long stringToInteger<long long>(const std::string& str, bool skipPartialMatchWarning);
 * unsigned long long stringToInteger<unsigned long long>(const std::string& str, bool skipPartialMatchWarning);
 */
```